### PR TITLE
Test opam 2.2

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

**Comment by hhugo** [comment]
@dra27, @kit-ty-kate, there seems to be some issue with git submodules in pinned packages on windows. Do you know anything about this ?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -82,27 +82,13 @@ jobs:
...
+        uses: kit-ty-kate/setup-ocaml@win52
         with:
           ocaml-compiler: ${{ matrix.ocaml-compiler }}
-          opam-repositories: |
-            dra27: https://github.com/dra27/opam-repository.git#windows-5.0
-            default: https://github.com/ocaml-opam/opam-repository-mingw.git#sunset
-            opam: https://github.com/ocaml/opam-repository.git
           dune-cache: true
           opam-depext: ${{ !matrix.skip-test }}
           opam-depext-flags: --with-test
```
Comment (smorimoto) [inline .github/workflows/build.yml:90]:
```suggestion
```

**Comment by smorimoto** [comment]
@hhugo After adding a temporary hack for Windows, only a few dependency errors remained.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -75,34 +75,25 @@ jobs:
...
 
       - name: Set-up Node.js
         uses: actions/setup-node@v4
         with:
           node-version: lts/*
 
       - name: Set-up OCaml ${{ matrix.ocaml-compiler }}
-        if: runner.os == 'Windows'
-        uses: ocaml/setup-ocaml@v2
+        uses: kit-ty-kate/setup-ocaml@win52
```
Comment (smorimoto) [inline .github/workflows/build.yml:87]:
```suggestion
        uses: ocaml/setup-ocaml@v3.0.0-alpha
```

**Comment by hhugo** [comment]
The issue should be fixed with https://github.com/ocaml/opam/pull/6046 and a new release of opam 2.2

**Pull request merged**